contracts/liquidity_hub/fee_collector/src/commands.rs

```diff
@@ -170,17 +171,13 @@ pub fn update_config(
 /// Aggregates the fees collected into the given asset_info.
 pub fn aggregate_fees(
     mut deps: DepsMut,
-    info: MessageInfo,
     env: Env,
-    ask_asset_info: AssetInfo,
     aggregate_fees_for: FeesFor,
 ) -> Result<Response, ContractError> {
     let config: Config = CONFIG.load(deps.storage)?;
 
-    // only the owner or the contract itself can aggregate the fees
-    if info.sender != config.owner && info.sender != env.contract.address {
-        return Err(ContractError::Unauthorized {});
-    }
+    // query fee collector to get the distribution asset
+    let ask_asset_info = query_distribution_asset(deps.as_ref())?;
```

Comment:
This change doesn't affect that since the contract will try to swap the assets if there's a swap route for it. In case it doesn't it will just skip the buyback for that particular asset, no contract errors or anything so we are good.